add self-signed certificate generation (#38)
neo4j/docs-ops-manager: modules/ROOT/pages/installation/self-signed-certificate.adoc

:description: This section contains instructions to generate a self-signed certificate for test purposes.
= Self-Signed Certificate Generation

[IMPORTANT]
====
Self-signed certificates are not recommended to be used in production environments.
For production environments, it is advisable to use a trusted certificate issuer.
This section outlines a practical way to generate a self-signed certificate for test and demo purposes.
====

The NOM Server contains a utility to generate a self-signed certificate suitable for your environment.

To invoke the utility and get help on usage, navigate to the NOM Server folder and execute the following command:

[source, terminal, role=noheader]
----
$> java -jar ./lib/server.jar ssc --help

utility to generate self signed certificate for TEST purposes only

sample usage:

to generate a self signed certificate for 'localhost', which could either be
accessed through DNS names of 'localhost.localdomain' or 'my.custom.domain', or
with IP addresses of '127.0.0.1' or '192.168.100.5';

$> java -jar ./lib/server.jar ssc -n localhost -o /tmp -p changeit \
   -d localhost.localdomain,my.custom.domain \
   -i 127.0.0.1,192.168.100.5

> it will create /tmp/localhost.cer and /tmp/localhost.pfx files upon completion.

Reviewer: ```suggestion
it creates /tmp/localhost.cer and /tmp/localhost.pfx files upon completion.
```

options (required options marked with *):
  -d, --dns=<dns>[,<dns>...]

>                             list of dns names you will use to access the

Reviewer: ```suggestion
                            list of dns names to use to access the
```

Reviewer 2: I am not sure this conveys exactly the right meaning. The user is supplying the dns names that they will use to access the server. 

The suggested change implies that somehow this will _cause_ the dns names to be used

Maybe something like : 
`list of dns names that are used to access the`

                              server, eg. --dns=localhost.localdomain,test.
                              local.domain

>   -i, --ip=<ip>[,<ip>...]   list of IP addresses you will use to access the

Reviewer: ```suggestion
  -i, --ip=<ip>[,<ip>...]   list of IP addresses to use to access the
```

                              server, eg. --ip=127.0.0.1,192.168.100.55

> * -n, --name=<name>         common name to be used in generated certificate,

Reviewer: ```suggestion
* -n, --name=<name>         common name to use in generated certificate,
```

                              eg. --name=localhost
* -o, --output=<output>     target directory, eg. --output=.
* -p, --password=<password> password for generated PKCS12 file
  -h, --help                help
----

If you are generating a certificate for running the server on `localhost`, `localhost` is the primary name set as Subject Name field on the certificate and is also used as the name of generated files.
For the certificate to be identified as valid, it also needs to include alternative names used to access the server.

> Assume there will also be a number of virtual machines that access the server through IP addresses `192.168.10.1` and `172.16.10.1`.

Reviewer: ```suggestion
Assume there are also a number of virtual machines that access the server through IP addresses `192.168.10.1` and `172.16.10.1`.
```

> Furthermore, a local DNS alias `nom.example.com` will be set up for `localhost`.

Reviewer: ```suggestion
Furthermore, a local DNS alias `nom.example.com` is set up for `localhost`.
```

> = Self-Signed Certificate Generation

Reviewer: Basically, description first, then header.

sample usage:

to generate a self signed certificate for 'localhost', which could either be
accessed through DNS names of 'localhost.localdomain' or 'my.custom.domain', or
with IP addresses of '127.0.0.1' or '192.168.100.5';

$> java -jar ./lib/server.jar ssc -n localhost -o /tmp -p changeit \
   -d localhost.localdomain,my.custom.domain \
   -i 127.0.0.1,192.168.100.5

it will create /tmp/localhost.cer and /tmp/localhost.pfx files upon completion.

options (required options marked with *):
  -d, --dns=<dns>[,<dns>...]
                            list of dns names you will use to access the
                              server, eg. --dns=localhost.localdomain,test.
                              local.domain
  -i, --ip=<ip>[,<ip>...]   list of IP addresses you will use to access the
                              server, eg. --ip=127.0.0.1,192.168.100.55
* -n, --name=<name>         common name to be used in generated certificate,
                              eg. --name=localhost
* -o, --output=<output>     target directory, eg. --output=.
* -p, --password=<password> password for generated PKCS12 file
  -h, --help                help
----

> Let's assume we are generating a certificate for running the server on `localhost`.

Reviewer: ```suggestion
If you are generating a certificate for running the server on `localhost`, `localhost` is the primary name set as Subject Name field on the certificate and is also used as the name of generated files.
```

> `localhost` is the primary name that will be set as Subject Name field on the certificate and will also be used as name of generated files.

Reviewer: ```suggestion

```

> For the certificate to be identified as valid, it also needs to include alternative names that we will use to access the server.

Reviewer: ```suggestion
For the certificate to be identified as valid, it also needs to include alternative names used to access the server.
```

> Assume that we also have a bunch of virtual machines that will access the server through IP addresses `192.168.10.1` and `172.16.10.1`.

Reviewer: ```suggestion
Assume you also have a number of virtual machines that access the server through IP addresses `192.168.10.1` and `172.16.10.1`.
```

> Furthermore, we setup a local DNS alias `nom.example.com` for our `localhost`.

Reviewer: ```suggestion
Furthermore, set up a local DNS alias `nom.example.com` for your `localhost`.
```

Author: I am not sure about this, we are continuing our assumptions here. I think we need to find an alternative way to explain it - like `assume that we set up a ...`? wdyt?

Reviewer: You can absolutely keep the assumption-part, but use _you_ instead of _we_. So, "assume that you set up a...".

> In order to generate a self-signed certificate for the above example, we will need to execute the following command;

Reviewer: ```suggestion
In order to generate a self-signed certificate for the above example, execute the following command;
```

[source, terminal, role=noheader]
----
java -jar ./lib/server.jar ssc -n localhost \
	-o ./certificates \
	-p changeit \
	-d nom.example.com \
	-i 192.168.10.1,172.16.10.1
----

> It will generate a key pair and a self-signed certificate and create `localhost.cer` and `localhost.pfx` files inside `./certificates` directory.

Reviewer: ```suggestion
It generates a key pair and a self-signed certificate and creates `localhost.cer` and `localhost.pfx` files inside `./certificates` directory.
```

> `localhost.pfx` will be assigned the password `changeit` which is provided to the command as an argument.

Reviewer: ```suggestion
`localhost.pfx` is assigned the password `changeit` which is provided to the command as an argument.
```
